src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -525,6 +527,103 @@ std::shared_ptr<LQPExpression> AbstractLQPNode::adapt_expression_to_different_lq
   return expression;
 }
 
+std::optional<std::pair<std::shared_ptr<const AbstractLQPNode>, std::shared_ptr<const AbstractLQPNode>>>
+AbstractLQPNode::find_subplan_mismatch(const std::shared_ptr<const AbstractLQPNode>& rhs) const {
+  return _find_subplan_mismatch_impl(shared_from_this(), rhs);
+}
+
+std::optional<std::pair<std::shared_ptr<const AbstractLQPNode>, std::shared_ptr<const AbstractLQPNode>>>
+AbstractLQPNode::_find_subplan_mismatch_impl(const std::shared_ptr<const AbstractLQPNode>& lhs,
+                                             const std::shared_ptr<const AbstractLQPNode>& rhs) {
+  if (lhs == rhs) return std::nullopt;
+  if (static_cast<bool>(lhs) != static_cast<bool>(rhs)) return std::make_pair(lhs, rhs);
+  if (!static_cast<bool>(lhs) && !static_cast<bool>(rhs)) return std::nullopt;
```

Comment:
Isn't this covered by `if (lhs == rhs) return std::nullopt;`?

Reply:
yes